dataprofiler/profilers/profile_builder.py

```diff
@@ -444,6 +445,477 @@ def clean_data_and_get_base_stats(df_series, sample_size,
+            if self.encoding == other.encoding else 'multiple files'
+        merged_profile.file_type = self.file_type \
+            if self.file_type == other.file_type else 'multiple files'
+        merged_profile._empty_line_count = (
+            self._empty_line_count + other._empty_line_count)
+        merged_profile.total_samples = self.total_samples + other.total_samples
+
+        samples = list(dict.fromkeys(self.sample + other.sample))
+        merged_profile.sample = random.sample(list(samples),
+                                              min(len(samples), 5))
+        merged_profile._profile = self._profile + other._profile
+        return merged_profile
+
+    def _get_sample_size(self, data):
+        """
+        Determines the minimum sampling size for detecting column type.
+
+        :param data: data to be profiled
+        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame]
+        :return: integer sampling size
+        :rtype: int
+        """
+        if self._samples_per_update:
+            return self._samples_per_update
+
+        len_data = len(data)
+        if len_data <= self._min_sample_size:
+            return int(len_data)
+        return max(int(self._sampling_ratio * len_data), self._min_sample_size)
+
+    def _update_base_stats(self, base_stats):
+        """
+        Updates the samples and line count of the class for the given dataset
+        batch.
+
+        :param base_stats: dictionary of basic sampling / data stats
+        :type base_stats: dict
+        :return: None
+        """
+        self.total_samples += base_stats["sample_size"]
+        self.sample = base_stats["sample"]
+        self._empty_line_count += base_stats["empty_line_count"]
+
+    @property
+    def profile(self):
+        """
+        Returns the UnstructuredCompiler which compiles the low level
+        inspectors.
+
+        :return: None
+        """
+        return self._profile
+
+    def report(self, report_options=None):
+        """
+        Returns the unstructured report based on all profiled data fed into the
+        profiler. User can specify the output_formats: (pretty, compact,
+        serializable, flat).
+            Pretty: floats are rounded to four decimal places, and lists are
+                shortened.
+            Compact: Similar to pretty, but removes detailed statistics such as
+                runtimes, label probabilities, index locations of null types,
+                etc.
+            Serializable: Output is json serializable and not prettified
+            Flat: Nested output is returned as a flattened dictionary
+
+        :var report_options: optional format changes to the report
+            `dict(output_format=<FORMAT>)`
+        :type report_options: dict
+        :return: dictionary report
+        :rtype: dict
+        """
+        if not report_options:
+            report_options = {
+                "output_format": None,
```

Comment:
Should you also set `"omit_keys"` to `[]` here?

Reply:
good call, just removing instead.

Reply:
Why? Are they not needed here?

Reply:
o wait, i was thinking that they were for omitting columns, i can re-add

Reply:
Wait why do they get set as `None` here but default to `[]` a few lines lower? Shouldn't this behavior be consistent?